README.md

```diff
@@ -28,10 +28,18 @@ The Caddyfile directive would look something like this:
 		bucket <bucket_name>
 		region <region_name>
 		index  <list of index file names>
-                endpoint <alternative S3 endpoint>
+		endpoint <alternative S3 endpoint>
 		root   <key prefix>
 		enable_put
 		enable_delete
+
+		# error_page allows you to use a custom error page depending on the HTTP status of the response.
```

Comment:
Instead of having a couple of directives for this one feature.  Maybe we could do something like this:
```
status_pages {
    404 s3://bucket/stuff/404.html
    default s3://bucket/stuff/error.html
}
```

Maybe have a short cut for the default case:
```
status_pages s3://bucket/stuff/error.html
```

I think something like this would be cleaner to document and use as an end user.  It It puts all the concerns around the status pages in just one directive.  Also, the README part should be more clear what that URL can look like.

Reply:
I agree that having duplicate directives like this doesn't seem ideal at first glance. I also thought of a nested directive, but I found it even more confusing for two reasons:
- In most languages I know, `404` is not a valid identifier because it starts with a digit. I'm not sure if that's also the case for the Caddyfile lexer (probably not), but it kind of hurts to read IMO.
- `404` itself (and every status code) becomes a directive which as don't think is perfectly correct and would maybe be clumsy to parse (as we couldn't use the usual switch/case).

As a counterargument, you could see the `error_page` directive more like [the `rewrite [matcher] <to>` directive](https://caddyserver.com/docs/caddyfile/directives/rewrite), where the matcher would just be a status code instead of a plain caddy matcher. So I think a syntax that would be more consistent with the "official" Caddyfile syntax would be something like `error_page [status] <s3 key>`:

```
error_page 400 "_400.html"
error_page 404 "_404.html"
error_page "_default_error.html"
```

I also saw you renamed `error_page` to `status_page` in your snippet. Do you have a strong preference for `status_page`? (I personally don't have a strong opinion on this, but I still reckon `error_page` is slightly more explicit on what it tries to achieve to the end user)

Do you think we could agree on the `error_page [status] <s3 key>` syntax?

Thanks for your time :heart: